multibranch/src/main/java/org/jenkinsci/plugins/workflow/multibranch/BranchJobProperty.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2015 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.workflow.multibranch;
+
+import hudson.Extension;
+import hudson.model.Item;
+import hudson.model.JobPropertyDescriptor;
+import hudson.security.ACL;
+import hudson.security.Permission;
+import javax.annotation.Nonnull;
+import jenkins.branch.Branch;
+import org.acegisecurity.Authentication;
+import org.jenkinsci.plugins.workflow.job.WorkflowJobProperty;
+
+/**
+ * Marker for jobs based on a specific branch.
+ */
+public class BranchJobProperty extends WorkflowJobProperty {
+
+    private /*@Nonnull once initialized */ Branch branch;
+
+    public synchronized Branch getBranch() {
+        return branch;
+    }
+
+    synchronized void setBranch(@Nonnull Branch branch) {
+        branch.getClass();
```

Comment:
My preferred style is to ensure that non-null parameters actually are... as other users may not be running findbugs and if they pass a null then all manner of issues can arise.

Some time back, the fastest way to check for a null was to just call `getClass()`. Current benchmarking of the JVM shows that this is no longer the case as of at least more recent builds of JVM 7. The only reason to use it though is on Java 1.6 codebases it does not leak either a branch point (which `if (x==null) throw NPE;` would do) or require Java 1.7 where `Objects.requireNonNull(x)` is the fastest.